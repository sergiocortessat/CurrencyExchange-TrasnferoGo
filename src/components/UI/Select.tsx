import React from "react";
import { MenuItem, Select as MaterialSelect } from "@material-ui/core";
import { currencyImages } from "../../data/currencies";

interface SelectProps {
  value: string;
  onChange: (e: React.ChangeEvent<{ value: unknown }>) => void;
  values: string[];
  className?: string;
  id?: string;
}

const Select: React.FC<SelectProps> = (props) => {
  const { value, onChange, values, className = "", id } = props;
  return (
    <MaterialSelect
      id={id}
      value={value}
      onChange={onChange}
      className={className}
    >
      {values.map((option) => (
        <MenuItem key={option} value={option}>
          <img src={currencyImages[option]} alt={option} />
          {option}
        </MenuItem>
      ))}
    </MaterialSelect>
  );
};

export default Select;
