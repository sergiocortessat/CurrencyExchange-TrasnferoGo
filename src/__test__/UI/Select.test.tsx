import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Select from '../../components/UI/Select';

describe('Select component', () => {
  test('renders options correctly', () => {
    const values = ['Option 1', 'Option 2', 'Option 3'];
    const onChange = jest.fn();
    const value = 'Option 2';
    
    const { asFragment } =  render(
      <Select values={values} onChange={onChange} value={value} />
    );
    expect(asFragment()).toMatchSnapshot();
    const selectElement = screen.getByRole('img', { name: value });
    expect(selectElement).toBeInTheDocument();
  });
});
