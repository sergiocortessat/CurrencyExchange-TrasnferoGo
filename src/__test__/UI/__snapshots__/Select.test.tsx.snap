// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders options correctly 1`] = `
<DocumentFragment>
  <div
    class="MuiInputBase-root MuiInput-root MuiInput-underline"
  >
    <div
      aria-haspopup="listbox"
      class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
      role="button"
      tabindex="0"
    >
      <img
        alt="Option 2"
      />
      Option 2
    </div>
    <input
      aria-hidden="true"
      class="MuiSelect-nativeInput"
      tabindex="-1"
      value="Option 2"
    />
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSelect-icon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
  </div>
</DocumentFragment>
`;
